e2e_tests/suite/test_smoke.py

```diff
@@ -0,0 +1,114 @@
+import time
+
+import pytest
+import requests
+import json
+
+from botocore.waiter import WaiterModel, create_waiter_with_client
+
+from e2e_tests.settings import NGINX_API_VERSION, RECONFIGURATION_DELAY
+
+
+def wait_for_changes_in_api(req_url, desired_capacity) -> None:
+    resp = requests.get(req_url)
+    nginx_upstream = json.loads(resp.text)
+    counter = 0
+    while len(nginx_upstream['peers']) != desired_capacity and counter < 10:
+        time.sleep(RECONFIGURATION_DELAY)
+        counter = counter + 1
+        resp = requests.get(req_url)
+        nginx_upstream = json.loads(resp.text)
+
+
+def wait_for_changes_in_aws(autoscaling_client, group_name, desired_capacity) -> None:
+    waiter_name = 'autoscaling_completed'
+    argument = f"contains(AutoScalingGroups[?(starts_with(AutoScalingGroupName, `{group_name}`) == `true`)]." \
+               f"[length(Instances[?LifecycleState=='InService']) == `{desired_capacity}`][], `false`)"
+    waiter_config = {
+        "version": 2,
+        "waiters": {
+            "autoscaling_completed": {
+                "acceptors": [
+                    {
+                        "argument": argument,
+                        "expected": True,
+                        "matcher": "path",
+                        "state": "success"
+                    },
+                    {
+                        "argument": argument,
+                        "expected": False,
+                        "matcher": "path",
+                        "state": "retry"
+                    }
+                ],
+                "delay": 5,
+                "maxAttempts": 20,
+                "operation": "DescribeAutoScalingGroups"
+            }
+        }
+    }
+    waiter_model = WaiterModel(waiter_config)
+    custom_waiter = create_waiter_with_client(waiter_name, waiter_model, autoscaling_client)
+    custom_waiter.wait()
+
+
+def scale_aws_group(autoscaling_client, group_name, desired_capacity) -> dict:
+    counter = 0
+    while counter < 10:
+        try:
+            response = autoscaling_client.set_desired_capacity(
+                AutoScalingGroupName=group_name,
+                DesiredCapacity=desired_capacity,
+                HonorCooldown=True,
+            )
+            print(f"Scaling activity started: {response}")
+            return response
+        except autoscaling_client.exceptions.ScalingActivityInProgressFault:
+            print("Scaling activity is in progress, wait for 60 seconds then retry.")
+            counter = counter + 1
+            time.sleep(RECONFIGURATION_DELAY)
+    pytest.fail(f"Failed to scale the group {group_name}")
+
+
+class TestSmoke:
+    @pytest.mark.parametrize("test_data", [
```

Comment:
will these tests be executed sequentially or in parallel? 

I think it is safe to execute them in parallel. That would speed up the running time.

Reply:
Yes, it would be nice to execute them in parallel. I kept that for the future not sure it's a must have for this very PR.